src/bun.js/webcore/ResumableSink.zig

```diff
@@ -0,0 +1,161 @@
+pub fn ResumableSink(
```

Comment:
I suggest adding a big comment here about how this is different from our other sink implemenetation.